core/src/main/java/org/fao/geonet/kernel/datamanager/IMetadataManager.java

```diff
@@ -69,6 +69,15 @@ public interface IMetadataManager {
      */
     void deleteMetadata(ServiceContext context, String metadataId) throws Exception;
 
+    /**
+     * Delete the record with the id metadataId and additionally take care of cleaning up resources, send events, ...
+     *
+     * @param context
+     * @param metadataId
+     * @throws Exception
+     */
+    void purgeMetadata(ServiceContext context, String metadataId, boolean withBackup) throws Exception;
```

Comment:
I would at least update the docs for `deleteMetadata` to indicate that doesn't send events and doesn't do a full cleanup and should be used in harvesters or similar code, otherwise use `purgeMetadata`.

Maybe can be good also to rename `deleteMetadata` to other name like `deleteMetadataWithoutEvents` and `purgeMetadata` to `deleteMetadata`.

Reply:
Added some doc on `deleteMetadata`. Will keep naming as it is for now.